### ISSUE TITLE
Ability to easily jump to the beginning of any visible word in the active buffer

### ISSUE THREAD

**Issue by damoonrashidi** [zed-industries/feedback#325]
**Before you begin**
I tried searching for jump / cursor etc but didn't find anything relevant, so hopefully this isn't a dupe, sorry if it is. Also I recognise that this is a pretty low priority request!

**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
I'd love for a way to move the cursor to the beginning of any visible word in the active buffer using only keyboard shortcuts.

**Describe alternatives you've considered**
The closest thing currently available is hitting `cmd+shift+o` to open the outline viewer and go to a declaration in close proximity to where you want to go to minimise the number of arrow key presses necessary. Unfortunately this doesn't work if there isn't a declaration close to where you want to jump.

Another alternative is hitting `ctrl+g` and jump to the beginning of specific line and then hitting `⌥ + →` until you hit your destination.

**Screenshots**
There are extensions for sublime/vs code that achieve this by overlaying a character sequence over the every word when a keyboard shortcut is hit as show below.

before hitting `cmd+shift+;`
![image](https://user-images.githubusercontent.com/207421/179216639-7c488710-80cc-4dab-bff4-efdd15b05a06.png)

after hitting `cmd+shift+;`
![image](https://user-images.githubusercontent.com/207421/179216675-c9f3d42c-43f6-496a-905d-60c35e4d10ff.png)

after typing any highlighted character / pair of characters (e.g. `u` `vk` `nd` etc) the cursor jumps to the highlighted location.

**Comment by kungfusheep** [comment]
Great suggestion - A good working reference for this is the AceJump plugin for sublime. This is the thing I miss most when I'm using zed, currently.

**Comment by damoonrashidi** [comment]
> Great suggestion - A good working reference for this is the AceJump plugin for sublime. This is the thing I miss most when I'm using zed, currently.

Ye sorry, forgot to mention, AceJump for sublime and EasyMotion for VS Code are some editor specific extensions that do this.

**Comment by rkusa** [comment]
Big +1! If it helps, here is another example. Amp has this built-in and calls it Jump Mode: https://amp.rs/docs/usage/#jump-mode

**Comment by alevroub** [comment]
Doesn't this seem way out of scope at this point in Zed's development? In terms of code navigation a better start would be further improvements to the vim mode.

**Comment by kungfusheep** [comment]
It depends on the target audience - does Zed simply wish to emulate vim or does it wish to define itself as a serious contender in the text editor market with its own ideals, benefits and control schemes? 

Personally if I wanted to use something that controlled like vim I would use vim, or neovim, for example. 

I’m here testing Zed because the developers genuinely seemed enthusiastic about creating something special. If we just aim for “sublime but better plugins” or “vscode but fast” then I feel it will have been a missed opportunity.

**Comment by JosephTLyons** [comment]
> Doesn't this seem way out of scope at this point in Zed's development? In terms of code navigation a better start would be further improvements to the vim mode.



> Doesn't this seem way out of scope at this point in Zed's development? In terms of code navigation a better start would be further improvements to the vim mode.

I agree with you - having foundational Vim stuff is going to help Vim users transfer over - as most Vim users are going to expect Vim-like behavior.  The team is working on Vim stuff all the time.

**Comment by damoonrashidi** [comment]
> Doesn't this seem way out of scope at this point in Zed's development?

Ye for sure at this point, just throwing it out there to see if there is any chance further down the line, it's a very reasonable take that this is a low priority feature right now, if it should be implemented at all.
